### PR TITLE
feat(INJI-202): hash uin/vid to store in mmkv and file storage

### PR DESCRIPTION
**Pull request by PoojaBabusing**

### CODE REVIEW

File: shared/storage.ts [hunk 1]
```diff
@@ -183,4 +183,10 @@ const getVCKeyName = (key: string) => {
   return key.split(':').splice(0, 4).join(':');
 };
 
+// To print the MMKV data cal this function in getItem
+// const getMMKVData = async () =>{
+//   const mmkvData = await MMKV.indexer.getKeys()
+//   console.log("mmkv data", mmkvData)
+// }
+
```
Comment (swatigoel) [inline shared/storage.ts:191]: why this commented code?

File: screens/Home/MyVcs/AddVcModalMachine.ts [hunk 2]
```diff
@@ -349,6 +368,15 @@ export const AddVcModalMachine =
           );
           return response.response.requestId;
         },
+
+        calculateHashedId: async (context) => {
+          const salt =
+            '1234567891011121314151617181920212223242526272829303132333435363';
```
Comment (swatigoel) [inline screens/Home/MyVcs/AddVcModalMachine.ts:374]: Can we move salt to constant or argon2iConfigForUinVid?

File: machines/vcItem.ts [hunk 3]
```diff
@@ -1331,14 +1349,15 @@ export const createVcItemMachine = (
   serviceRefs: AppServices,
   vcKey: string
 ) => {
-  const [, idType, id, requestId, isPinned] = vcKey.split(':');
+  const [, idType, hashedId, requestId, isPinned, id] = vcKey.split(':');
```
Comment (swatigoel) [inline machines/vcItem.ts:1352]: is `id` at the end in UIN or VID?

File: shared/storage.ts (hunk 1 above)
Comment (PoojaBabusing) [reply shared/storage.ts:191]: This function is to get the data from MMKV. If anybody wants to check the MMKV data they can uncomment this and use it.

File: machines/vcItem.ts (hunk 3 above)
Comment (PoojaBabusing) [reply machines/vcItem.ts:1352]: yes, as we have changed the  " VC_ITEM_STORE_KEY =`vc:${vc.idType}:${vc.hashedId}:${vc.requestId}:${vc.isPinned}:${vc.id}`;  "
the id is at the end

File: screens/Home/MyVcs/AddVcModalMachine.ts (hunk 2 above)
Comment (PoojaBabusing) [reply screens/Home/MyVcs/AddVcModalMachine.ts]: committed this change

File: shared/storage.ts (hunk 1 above)
Comment (swatigoel) [reply shared/storage.ts:191]: Then, let's add it as uncommented one. No one is going to see commented code in future to uncomment and use

File: shared/constants.ts [hunk 4]
```diff
@@ -80,3 +80,6 @@ export const argon2iConfigForUinVid: Argon2iConfig = {
   hashLength: 5,
   mode: 'argon2i',
 };
+
+export const salt =
```
Comment (swatigoel) [inline shared/constants.ts:84]: name it agrdon2iSalt so that it's clear with name

Comment (PoojaBabusing) [reply shared/constants.ts]: renamed

File: shared/storage.ts (hunk 1 above)
Comment (PoojaBabusing) [reply shared/storage.ts:191]: uncommented it

File: shared/storage.ts [hunk 5]
```diff
@@ -184,9 +184,9 @@ const getVCKeyName = (key: string) => {
 };
 
 // To print the MMKV data cal this function in getItem
-// const getMMKVData = async () =>{
-//   const mmkvData = await MMKV.indexer.getKeys()
-//   console.log("mmkv data", mmkvData)
-// }
+const getMMKVData = async () => {
+  const mmkvData = await MMKV.indexer.getKeys();
+  console.log('mmkv data', mmkvData);
```
Comment (swatigoel) [inline shared/storage.ts:189]: remove console log

**Review by swatigoel — APPROVED** [review]

**Pull request merged**